reinforcement/algorithm/reinforce.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+
+
+def _calc_signal(ops, action_dist, returns):
+    return ops.reduce_mean(ops.log(action_dist) * returns)
```

Comment:
This is probably wrong, the action probs need to be combined with the actual selected action in some way